examples/nlp/token_classification/token_classification_train.py

```diff
@@ -101,15 +102,15 @@
 
 @hydra_runner(config_path="conf", config_name="token_classification_config")
 def main(cfg: DictConfig) -> None:
-    trainer = pl.Trainer(**cfg.trainer)
+    trainer = pl.Trainer(plugins=[NLPDDPPlugin()], **cfg.trainer)
     exp_manager(trainer, cfg.get("exp_manager", None))
 
     if not cfg.pretrained_model:
         logging.info(f'Config: {OmegaConf.to_yaml(cfg)}')
         model = TokenClassificationModel(cfg.model, trainer=trainer)
     else:
         if os.path.exists(cfg.pretrained_model):
-            model = TokenClassificationModel.restore_from(cfg.pretrained_model)
+            model = TokenClassificationModel.restore_from(cfg.pretrained_model, trainer=trainer)
         elif cfg.pretrained_model in TokenClassificationModel.get_available_model_names():
             model = TokenClassificationModel.from_pretrained(cfg.pretrained_model)
```

Comment:
so this case is guaranteed to be non-model parallel, right?

Reply:
No this can be model parallel or non-model parallel.

Reply:
I meant elif part where trainer instance isn't passed:
```python
elif cfg.pretrained_model in TokenClassificationModel.get_available_model_names():
            model = TokenClassificationModel.from_pretrained(cfg.pretrained_model)
```

Reply:
There's no model parallel models on NGC yet...

Reply:
I meant for the future. Better to have it in from_pretrained at the ModelPT level and then forward the trainer to the actual restore_from call inside.